types/index.d.ts

```diff
@@ -229,6 +235,143 @@ declare module '@azure/functions' {
+         * Has the same functionality as getHeader
+         * Get the value of a particular header field
+         */
+        get(field: string): any;
+        /**
+         * Removes a particular header field
+         * @returns the updated HttpResponseApi instance
+         */
+        removeHeader(field: string): HttpResponseApi;
+        /**
+         * Set the 'Content-Type' header to a particular value
+         * @returns the updated HttpResponseApi instance
+         */
+        type(type: string): HttpResponseApi;
+        /**
+         * Automatically sets the content-type then calls context.done()
+         * @returns updated HttpResponseApi instance
+         * @deprecated this method calls context.done() which is deprecated, use async/await and pass the response as the return value instead.
+         * See the docs here for more information: https://aka.ms/functions-js-async-await
+         */
+        send(body?: any): HttpResponseApi;
+        /**
+         * Same as send()
+         * Automatically sets the content-type then calls context.done()
+         * @returns updated HttpResponseApi instance
+         * @deprecated this method calls context.done() which is deprecated, use async/await and pass the response as your function's return value instead.
+         * See the docs here for more information: https://aka.ms/functions-js-async-await
+         */
+        end(body?: any): HttpResponseApi;
+        /**
+         * Sets the status code then calls send()
+         * @returns updated HttpResponseApi instance
+         * @deprecated this method calls context.done() which is deprecated, use async/await and pass the response as your function's return value instead.
+         * See the docs here for more information: https://aka.ms/functions-js-async-await
+         */
+        sendStatus(statusCode: string | number): HttpResponseApi;
+        /**
+         * Sets the 'Content-Type' header to 'application/json' then calls send(body)
+         * @deprecated this method calls context.done() which is deprecated, use async/await and pass the response as your function's return value instead.
+         * See the docs here for more information: https://aka.ms/functions-js-async-await
+         */
+        json(body?: any): void;
+    }
+    /**
+     * Http response object.
+     * This is not the default on the Context object, but you may replace context.res with an object of this type when using HTTP triggers.
+     */
+    export interface HttpResponseObject {
+        /**
+         * HTTP response headers.
+         */
+        headers?: HttpResponseHeaders;
+        /**
+         *  HTTP response cookies.
+         */
+        cookies?: Cookie[];
+        /**
+         * HTTP response body.
+         */
+        body?: any;
+        /**
+         * HTTP response status code.
+         * @default 200
+         */
+        statusCode?: number | string;
+        /**
+         * HTTP response status code
+         * @default 200
```

Comment:
Please add notes like this

to statusCode:
> This property takes priority over the `status` property

to status:
> The same as `statusCode`. This property is ignored if `statusCode` is set

Reply:
Done